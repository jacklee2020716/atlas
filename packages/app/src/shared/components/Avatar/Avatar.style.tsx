import { makeStyles, StyleFn } from '../../utils'
import { colors, spacing } from '../../theme'

export type AvatarStyleProps = {
  size?: 'small' | 'default' | 'large'
}

const container: StyleFn = (_, { size = 'default' }) => {
  const width = size === 'small' ? spacing.xs : size === 'default' ? spacing.m : spacing.xl
  return {
    borderRadius: 999,
    minWidth: width,
    backgroundColor: colors.gray[400],
    color: colors.white,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',

    '& > span': {
      textTransform: 'uppercase',
      fontSize: '0.875rem',
      lineHeight: 1.43,
    },
  }
}

const img: StyleFn = () => ({
  width: '100%',
  height: '100%',
  objectFit: 'cover',
  borderRadius: 999,
})

export const useCSS = (props: AvatarStyleProps) => ({
  container: makeStyles([container])(props),
  img: makeStyles([img])(props),
})
