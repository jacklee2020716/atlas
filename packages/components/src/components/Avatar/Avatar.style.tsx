import { css } from "@emotion/core";
import { spacing, colors } from "../../theme";

export type AvatarStyleProps = {
	img?: string;
	size?: "small" | "default" | "large";
};

export let makeStyles = ({ img, size = "default" }: AvatarStyleProps) => {
	let width = size === "small" ? spacing.xs : size === "default" ? spacing.m : spacing.xl;
	return css`
		background-image: ${img ? `url(${img})` : `radial-gradient(${colors.gray[500]}, ${colors.gray[500]})`};
		background-size: cover;
		background-position: center;
		border-radius: 999px;
		min-width: ${width};
		min-height: ${width};
	`;
};
