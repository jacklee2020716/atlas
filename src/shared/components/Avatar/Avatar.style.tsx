import { makeStyles, StyleFn } from '../../utils'
import { colors, sizes } from '../../theme'

export type AvatarStyleProps = {
  size?: 'small' | 'default' | 'large'
}

const container: StyleFn = (_, { size = 'default' }) => {
  const width = size === 'small' ? sizes(2) : size === 'default' ? sizes(4) : sizes(6)
  return {
    borderRadius: 999,
    minWidth: width,
    backgroundColor: colors.gray[400],
    color: colors.white,
    display: 'inline-flex',
    justifyContent: 'center',
    alignItems: 'center',

    '& > span': {
      textTransform: 'uppercase',
      fontSize: '0.875rem',
      lineHeight: 1.43,
    },
  }
}

const img: StyleFn = () => ({
  width: '100%',
  height: '100%',
  objectFit: 'cover',
  borderRadius: 999,
})

export const useCSS = (props: AvatarStyleProps) => ({
  container: makeStyles([container])(props),
  img: makeStyles([img])(props),
})
