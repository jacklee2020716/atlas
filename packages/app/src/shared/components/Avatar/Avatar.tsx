import React from 'react'
import { SerializedStyles } from '@emotion/core'
import { AvatarStyleProps, useCSS } from './Avatar.style'

export type AvatarProps = {
  onClick: (e: React.MouseEvent<HTMLElement>) => void
  outerStyles?: SerializedStyles
  img: string
  name: string
  className?: string
} & AvatarStyleProps

function initialsFromName(name: string): string {
  const vowels = ['a', 'e', 'i', 'o', 'u', 'y']
  const [first = '', second = ''] = name.split('')
  return vowels.includes(second) ? first : `${first}${second}`
}

const Avatar: React.FC<Partial<AvatarProps>> = ({
  img,
  outerStyles,
  onClick = () => {},
  name,
  className,
  ...styleProps
}) => {
  const styles = useCSS({ ...styleProps })
  return (
    <div css={[styles.container, outerStyles]} className={className} onClick={onClick}>
      {img ? <img src={img} css={styles.img} /> : <span>{initialsFromName(name || '')}</span>}
    </div>
  )
}

export default Avatar
