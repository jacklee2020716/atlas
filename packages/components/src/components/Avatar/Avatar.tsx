import React from "react";
import { makeStyles, AvatarStyleProps } from "./Avatar.style";

export type AvatarProps = {} & AvatarStyleProps;

export default function Avatar({ ...styleProps }: AvatarProps) {
  let styles = makeStyles(styleProps);
  return (
    <div css={styles} />
  );
}
