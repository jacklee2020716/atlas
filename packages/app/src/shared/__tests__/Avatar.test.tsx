import React from 'react'
import { mount } from 'enzyme'
import Avatar from '@/shared/components/Avatar'
describe('Avatar component', () => {
  it('Should render avatar correctly', () => {
    expect(mount(<Avatar img="https://source.unsplash.com/WLUHO9A_xik/1600x900" />)).toMatchSnapshot()
  })
})
