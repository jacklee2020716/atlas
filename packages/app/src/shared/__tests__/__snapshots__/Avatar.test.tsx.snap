// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component Should render avatar correctly 1`] = `ReactWrapper {}`;
